app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/fragments/PopupAttributesListFragment.kt

```diff
@@ -43,11 +50,69 @@ class PopupAttributesListFragment : Fragment() {
             false
         )
 
-        val viewModel: IdentifyResultViewModel by activityViewModels()
+        val identifyResultViewModel: IdentifyResultViewModel by activityViewModels()
 
-        binding.identifyResultViewModel = viewModel
+        binding.identifyResultViewModel = identifyResultViewModel
+        binding.popupViewModel = popupViewModel
         binding.lifecycleOwner = this
 
+        popupViewModel.isPopupInEditMode.observe(viewLifecycleOwner, Observer {
+            popupView.setEditMode(it)
+        })
+
+        popupViewModel.toggleSavingPopupProgressBarEvent.observeEvent(viewLifecycleOwner) { isShowProgressBar ->
+            if (isShowProgressBar) {
+                progressBarLayout.visibility = View.VISIBLE
+            } else {
+                progressBarLayout.visibility = View.GONE
+            }
+        }
+
+        popupViewModel.showSavePopupErrorEvent.observeEvent(viewLifecycleOwner) { errorMessage ->
+            showAlertDialog(errorMessage)
+        }
+
+        popupViewModel.confirmCancelPopupEditingEvent.observeEvent(viewLifecycleOwner) {
+            showConfirmCancelEditingDialog()
+        }
+
         return binding.root
     }
+
+    /**
+     * Shows dialog to confirm cancelling edit mode on popup view.
+     */
+    private fun showConfirmCancelEditingDialog() {
+        val dialogBuilder = AlertDialog.Builder(requireContext())
+        dialogBuilder.setMessage("Discard changes?")
+            .setCancelable(false)
+            // positive button text and action
+            .setPositiveButton(getString(R.string.ok)) { dialog, id ->
+                popupViewModel.cancelEditing()
+            }
+            // negative button text and action
+            .setNegativeButton(getString(R.string.cancel)) { dialog, id -> dialog.cancel()
+            }
+        val alert = dialogBuilder.create()
+        // show alert dialog
+        alert.show()
+    }
+
+    /**
+     * Shows error message to the use in a dialog.
```

Comment:
user

Reply:
updated

Reply:
Some how I missed to push it to remote. Updated now.